Limitations
-----------


This will work
try
{
	exception throwing code
}
catch(...)
{
	...
}
[NDFiber yieldToFiber:a]

But this will NOT work
try
{
	[NDFiber yieldToFiber:a]
	exception throwing code
}
catch(...)
{
	...
}

¥ autorelease pools: autorelease pools work fine as long as you realise that after a yield any autoreleased object may have been released by an autorelease pool in the other fiber.

For example this can fail

	id	theObject = [obj autoreleasedObject];
	[NDFiber yieldToFiber:a]
	[theObject doSomething];		// object may longer be a valid object


The fix is to retain the object and release it after the yield

	id	theObject = [obj autoreleasedObject];
	[theObject retain];
	[NDFiber yieldToFiber:a]
	[theObject doSomething];
	[theObject release];
